### ISSUE TITLE
Import suggestion from crate not directly in Cargo.toml

### ISSUE THREAD

**Issue by bugadani** [rust-analyzer/rust-analyzer#7373]
I have a project that references `actix-web` and I'm trying to import a type (in VS Code using `ctrl-.` and selecting from the suggestion list).

Trying to import `HttpResponse`, the suggestion list contains `actix_http::Response`, which is the definition of the type I want, but I don't have `actix_http` referenced in my `Cargo.toml` so importing that doesn't compile. Instead, rust-analyzer should realize that `actix_web` reexports the type as `pub use actix_http::Response as HttpResponse;` and offer `actix_web::HttpResponse` instead.

TL;DR: import resolution should stop when encountering a reexport from an unreferenced crate.

**Comment by flodiebold** [comment]
Hmm that's somewhat surprising, since we do take care to handle this situation correctly and have [tests](https://github.com/rust-analyzer/rust-analyzer/blob/323138f32ea74cfe9f5381e9e170cf87e7592818/crates/hir_def/src/find_path.rs#L591-L601) for it. Can you provide a minimal reproduction? Does this happen with a simple crate just importing actix-web?

**Comment by bugadani** [comment]
This is on me, I have actix-http imported as dev dependency.

**Comment by flodiebold** [comment]
That would explain it. We could still handle that correctly if we analyzed the crate both with and without the test configuration, which we might have to do anyway; but I'll close this nevertheless to keep things clean.

**Comment by bugadani** [comment]
That's fine.

I wonder if it is feasible to offer suggestion alternatives from both dev and non-dev dependencies in this case?

**Comment by flodiebold** [comment]
Only by analyzing the crate in both configurations, I think. Which we might do, but it's not trivial :sweat_smile:

**Comment by bugadani** [comment]
I'm sorry, I didn't want you to have to repeat yourself. I meant something along the lines of: Offer reexports, but limit them to one per external crate. I don't have any knowledge of RA though, so if this also needs analyzing in both configs, sorry :P

Also, I think for a properly thorough solution RA would need to analyze in both configs AND with all the possible features enabled, and disabled, so this could get messy :)
